storops_test/unity/resource/test_nas_server.py

```diff
@@ -208,6 +210,53 @@ def f():
 
         assert_that(f, raises(UnitySmbNameInUseError, 'name already exists'))
 
+    @patch_rest
+    @ddt.data(
+        {'dst_nas': 'nas_12', 'replication_name': None, 'remote_system': None,
+         'expected_rep_name': 'rep_sess_nas_6_nas_12_local'},
+        {'dst_nas': 'nas_6', 'replication_name': 'remote-nas_6-nas_6',
+         'remote_system': 'RS_6', 'expected_rep_name': 'remote-nas_6-nas_6'},
+    )
+    @ddt.unpack
+    def test_replicate(self, dst_nas, replication_name, remote_system,
+                       expected_rep_name):
+        if remote_system:
+            remote_system = UnityRemoteSystem(_id=remote_system, cli=t_rest())
+        nas = UnityNasServer.get(cli=t_rest(), _id='nas_6')
+        rep_session = nas.replicate(dst_nas, 60,
+                                    replication_name=replication_name,
+                                    remote_system=remote_system)
+        assert_that(rep_session.name, equal_to(expected_rep_name))
+
+    @patch_rest
+    @ddt.data(
+        {'dst_nas_server_name': None, 'remote_system': None, 'rep_name': None,
+         'dst_sp': None, 'is_backup_only': None,
+         'expected_rep_session_id': '103079215112_APM00192210744_0000'
+                                    '_103079215117_APM00192210744_0000'},
+        {'dst_nas_server_name': 'nas-liangr', 'remote_system': 'RS_6',
+         'rep_name': 'nas-server-rep',
+         'dst_sp': NodeEnum.SPA, 'is_backup_only': False,
+         'expected_rep_session_id': '103079215112_APM00192210744_0000'
+                                    '_103079215110_FNM00184901113_0000'},
+    )
+    @ddt.unpack
+    def test_replicate_with_dst_resource_provisioning(self,
```

Comment:
Should add a test with remote_system specified but no dst_sp specified.

Reply:
Sure.